tests/interface/test_utilities.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+
+from vivarium.interface.utilities import get_output_model_name_string
+
+_MODEL_SPEC_STEM = "model_spec_name"
+_ARTIFACT_STEM = "artifact_name"
+_ARTIFACT_FROM_MODEL_SPEC_STEM = f"{_ARTIFACT_STEM}_from_model_spec"
+
+_ARTIFACT_PATH = Path(f"/totally/fake/path/for/artifact/{_ARTIFACT_STEM}.hdf")
+_MODEL_SPEC_ARTIFACT_PATH = Path(
+    f"/totally/fake/path/for/artifact/{_ARTIFACT_FROM_MODEL_SPEC_STEM}.hdf"
+)
+
+_MODEL_SPEC_CONTENTS_WITHOUT = """
+configuration:
+    input_data:
+        input_draw_number: 0
+"""
+
+_MODEL_SPEC_CONTENTS_WITH = (
+    _MODEL_SPEC_CONTENTS_WITHOUT
+    + f"""
+        artifact_path: '{_MODEL_SPEC_ARTIFACT_PATH}'
+"""
+)
+
+
+def _write_file(path: Path, contents: str):
+    with open(path, "w") as file:
+        file.write(contents)
+
+
+def test_get_output_model_name_string(tmp_path):
```

Comment:
It's better form to use `@pytest.mark.parametrize` to capture each of these three cases. See `test_artifact` line 526 for an example.